sqlglot/dialects/hive.py

```diff
@@ -18,6 +18,34 @@
 from sqlglot.parser import Parser, parse_var_map
 from sqlglot.tokens import Tokenizer
 
+# (FuncType, Multiplier)
+DATE_DELTA_INTERVAL = {
+    "YEAR": ("ADD_MONTHS", 12),
+    "MONTH": ("ADD_MONTHS", 1),
+    "QUARTER": ("ADD_MONTHS", 3),
+    "WEEK": ("DATE_ADD", 7),
+    "DAY": ("DATE_ADD", 1),
+}
+
+
+def _add_date(self, expression):
+    unit = expression.text("unit").upper()
+    func, multiplier = DATE_DELTA_INTERVAL.get(unit, ("DATE_ADD", 1))
+    modified_increment = (
+        int(expression.text("expression")) * multiplier if expression.expression.is_number else expression.expression
+    )
+    return f"{func}({self.sql(expression, 'this')}, {modified_increment})"
+
+
+def _date_diff(self, expression):
+    unit = expression.text("unit").upper()
+    sql_func = "MONTHS_BETWEEN" if unit == "MONTH" or unit == "QUARTER" else "DATEDIFF"
+    _, multiplier = DATE_DELTA_INTERVAL.get(unit, ("", 1))
+    if multiplier > 1:
+        return f"{sql_func}({self.sql(expression, 'this')}, {self.sql(expression, 'expression')}) / {multiplier}"
+    else:
+        return f"{sql_func}({self.sql(expression, 'this')}, {self.sql(expression, 'expression')})"
+
```

Comment:
Let's rename this to _date_diff_sql. Also, maybe it can be simplified to:
```python
def _date_diff_sql(self, expression):
    unit = expression.text("unit").upper()
    sql_func = "MONTHS_BETWEEN" if unit == "MONTH" or unit == "QUARTER" else "DATEDIFF"
    _, multiplier = DATE_DELTA_INTERVAL.get(unit, ("", 1))
    multiplier_sql = f" / {multiplier}" if multiplier > 1 else ""
    diff_sql = f"{sql_func}({self.sql(expression, 'this')}, {self.sql(expression, 'expression')})"
    return f"{diff_sql}{multiplier_sql}"
```